### ISSUE TITLE
Archives

### ISSUE THREAD

**Issue by DonaldTsang** [DangerOnTheRanger/maniwani#60]
DesuArchive, archived.org etc. are good for imageboards in general.
Also OpenIB has its own archiving tool. And archive.fo is a good supplement.
Lastly, HollaForums exists even if the sites are full of malware ads.

**Comment by DangerOnTheRanger** [comment]
The way I understand how tools like desuarchive and company work is that they scrape the site either after periodically polling for new content, or upon an API call that notifies the scraper of new content. If that's the case (and I could be wrong), then support for Maniwani installations is a feature that you'd have to go around and ask those specific archiving tools for.

I think something we could do without having to go through all of that would to simply have a link in each thread that, when clicked, generated an archive on a place like archive.fo or something like that. It wouldn't be searchable in the way a full-blown imageboard-specific archive would be, but it would be super lightweight, and something that could get implemented without needing to submit PRs to an external project (or write our own archiver).

**Comment by DonaldTsang** [comment]
Sounds good, as long as it is as easy as archive.fo (assuming ".fo" is open source) then it is useful

**Comment by DangerOnTheRanger** [comment]
Something else to keep in mind if we went the route of providing a link to an archive site like archive.fo (which keeps on returning HTTP 500s when I attempt to try it out, weird) is that images/other media would probably not get saved outside of thumbnails. I don't see any real solution to that as it stands, but I also don't personally consider that a reason not to implement an archive link.

**Comment by DonaldTsang** [comment]
Archiving images can be done in archive.fo as well

**Comment by DangerOnTheRanger** [comment]
You're right, although I don't think that site archives recursively, so people would have to archive attachments manually. I don't think that's a showstopper, but it's something to keep in mind. I also don't believe archive.fo would work on something like a 3D model or code (3D model uploading will be implemented with completion of #42, and code uploading/thumbnailing is done already), but I could be wrong about that.

**Comment by DonaldTsang** [comment]
@DangerOnTheRanger generally speaking the less JS, the better archive.fo can archive.
Check these two pieces of code
`'http://archive.today/?run=1&url='+encodeURIComponent(link)` (in JS)
`'http://archive.today/?run=1&url='+urllib.parse.quote(link)` (in Python)
